source/slang-capture-replay/capture_utility.cpp

```diff
@@ -0,0 +1,66 @@
+
+#include <cstdlib>
+#include <cstring>
+#include <string>
+#include <stdarg.h>
+#include <mutex>
+
+#include "capture_utility.h"
+
+constexpr const char* kCaptureLayerEnvVar = "SLANG_CAPTURE_LAYER";
+constexpr const char* kCaptureLayerLogLevel = "SLANG_CAPTURE_LOG_LEVEL";
+
+namespace SlangCapture
+{
+    static unsigned int g_logLevel = LogLevel::Silent;
+
+    bool isCaptureLayerEnabled()
+    {
+        if (const char* envVar = std::getenv(kCaptureLayerEnvVar))
+        {
+            if (std::strcmp(envVar, "1") == 0)
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    void setLogLevel()
+    {
+        static std::mutex logLevelMutex;
```

Comment:
I think a mutex locking is an overkill for setting an integer value.
Alternatively we can make the variable thread-local and let each thread have their own variants.

Reply:
Yes, thinking about the same thing as well.